overlord/devicestate/devicemgr.go

```diff
@@ -788,6 +788,30 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// canAutoRefresh is a helper that checks if the device is able to
+// auto-refresh
+func canAutoRefresh(st *state.State) bool {
+	// no need to wait for seeding on classic
+	if release.OnClassic {
+		return true
+	}
+
+	// on all-snap devices we need to be seeded first
+	var seeded bool
+	st.Get("seeded", &seeded)
+	if !seeded {
+		return false
+	}
+
+	// and we also need to have a serial
+	_, err := Serial(st)
```

Comment:
this is correct long term, otoh it means developer images with custom models will not auto-refresh atm